packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/UserImpl.kt

```diff
@@ -16,10 +16,60 @@
 
 package io.realm.mongodb.internal
 
+import io.realm.internal.interop.AppCallback
 import io.realm.internal.interop.NativePointer
+import io.realm.internal.interop.RealmInterop
 import io.realm.mongodb.User
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
 
 internal class UserImpl(
     val nativePointer: NativePointer,
-    val app: AppImpl
-) : User
+    override val app: AppImpl
+) : User {
+
+    override val state: User.State
+        get() = User.State.fromCoreState(RealmInterop.realm_user_get_state(nativePointer))
+
+    // TODO Can maybe fail, but we could also cache the return value?
+    override val identity: String
+        get() = RealmInterop.realm_user_get_identity(nativePointer)
+
+    override val loggedIn: Boolean
+        get() = RealmInterop.realm_user_is_logged_in(nativePointer)
+
+    override suspend fun logOut() {
+        return suspendCoroutine { continuation ->
```

Comment:
I have reworked the pattern for C-API async remote calls in #469 to ensure that requests are performed on a background thread. We should adopt it for all these remote calls.